packages/web/src/hooks/useAuthenticatedCallback.ts

```diff
@@ -12,13 +12,20 @@ const { getHasAccount } = accountSelectors
 
 export const useAuthenticatedCallback = <T extends (...args: any) => any>(
   callback: T,
-  deps: any[]
+  deps: any[],
+  stopPropagation = false
 ) => {
   const isSignedIn = useSelector(getHasAccount)
   const dispatch = useDispatch()
 
   return useCallback(
     (...args: Parameters<T>) => {
+      if (stopPropagation) {
+        // In the case we are clicking something that expects to
+        // propagate clicks, don't propagate
+        ;(args[0] as MouseEvent)?.stopPropagation()
```

Comment:
Couple of questions
- If `stopPropagation` is true but `args[0]` isn't a mouse event, this would error right?
- Do we ever want to _not_ stop propagation in the case of a `MouseEvent`?
- Would it be clearer (albeit more boilerplate) to explicitly wrap the callback returned by `useAuthenticatedCallback` in another callback that calls `stopPropagation`? This feels a bit more coupled than it needs to be. Maybe a `useAuthenticatedClickCallback` variant would be good?

Reply:
`useAuthenticatedClickCallback` did that- i like the idea